### DISCUSSION TITLE
[Feature Request]: 能不能写一个后台自动更新公网 ip到企业微信的插件

### DISCUSSION THREAD

**Discussion by xxww0098** [jxxghp/MoviePilot#3368] [讨论]
### 当前程序版本

1.1.5

### 功能改进

1、插件使用浏览器获取企业微信登入二维码
2、扫码登入获取企业微信 cookie、和目前的white_ip_list
3、后台定时动作，保活 cookie
4、当企业微信发送不了信息时，触发插件，自动使用 api更新追加white_ip_list

目前这个思路在于能不能保活 cookie

**Comment by thsrite** [comment]
貌似有cookie不能直接模拟请求设置可信ip接口
应该还有其他参数
你可以先模拟请求试试

**Comment by zraxx** [comment]
确实有需求！

**Comment by HuntzzZ** [comment]
你是懂偷懒的，哈哈哈

**Comment by jxxghp** [comment]
不确定Cookie能否保活，大佬们可以试试

**Comment by naoki66** [comment]
自动修改可信IP，你们真牛

**Comment by EkkoG** [comment]
如果手上有代理节点可以设置给企业微信 API 域名，因为代理节点出口一般不变所以不用频繁更新后台可信 IP，如果手上有 VPS，可以自己搭建让企业微信 API 域名流量走自己的 VPS，这样更稳，如果自己的节点不够稳定，手上又有稳定的机场节点，可以用代理链让自己的节点更稳定

以上都和本 issue 没什么太大的关系，只是提一下别的利用自己可以控制的资源来实现和企业微信 API 通信的方案

**Reply by pavelyuan** [reply]
用机场节点这是我没想到的，666
